homeassistant/components/watchyourlan/config_flow.py

```diff
@@ -0,0 +1,59 @@
+"""Config flow for WatchYourLAN integration."""
+
+import logging
+from typing import Any
+
+from httpx import ConnectError, HTTPStatusError
+import voluptuous as vol
+from watchyourlanclient import WatchYourLANClient
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_URL, CONF_VERIFY_SSL
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+# Schema for the user setup form
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_URL): str,
+        vol.Optional(CONF_VERIFY_SSL, default=False): bool,
+    }
+)
+
+
+class WatchYourLANConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for WatchYourLAN."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            try:
+                # Use the WatchYourLANClient to validate the connection
+                api_client = WatchYourLANClient(
+                    base_url=user_input[CONF_URL],
+                    async_mode=True,
+                    verify_ssl=user_input[CONF_VERIFY_SSL],
+                )
```

Comment:
Does creating the object raise?

Reply:
No, creating the client does raise. The client only raises when the call(s) to the API actually fail, which requires calling the client's methods.

Reply:
Okay so let's move it out of the try block